GameServerLib/GameObjects/Spell/Missile/SpellCircleMissile.cs

```diff
@@ -0,0 +1,264 @@
+using GameServerCore.Domain.GameObjects.Spell.Missile;
+using GameServerCore.Enums;
+using LeagueSandbox.GameServer.Content;
+
+namespace LeagueSandbox.GameServer.GameObjects.Spell.Missile
+{
+    public class SpellCircleMissile : SpellMissile, ISpellCircleMissile
+    {
+        // Function Vars.
+        private bool _atDestination;
+
+        public override MissileType Type { get; protected set; } = MissileType.Circle;
+        /// <summary>
+        /// Number of objects this projectile has hit since it was created.
+        /// </summary>
+        public List<IGameObject> ObjectsHit { get; }
+        /// <summary>
+        /// Position this projectile is moving towards. Projectile is destroyed once it reaches this destination. Equals Vector2.Zero if TargetUnit is not null.
+        /// </summary>
+        public Vector2 Destination { get; protected set; }
+
+        public SpellCircleMissile(
+            Game game,
+            int collisionRadius,
+            ISpell originSpell,
+            ICastInfo castInfo,
+            float moveSpeed,
+            SpellDataFlags overrideFlags = 0, // TODO: Find a use for these
+            uint netId = 0,
+            bool serverOnly = false
+        ) : base(game, collisionRadius, originSpell, castInfo, moveSpeed,  overrideFlags, netId, serverOnly)
+        {
+            // TODO: Verify if there is a case which contradicts this.
+            // Line and Circle Missiles are location targeted only.
+            TargetUnit = null;
+
+            Position = new Vector2(castInfo.SpellCastLaunchPosition.X, castInfo.SpellCastLaunchPosition.Z);
+
+            var goingTo = new Vector2(castInfo.TargetPositionEnd.X, castInfo.TargetPositionEnd.Z) - Position;
+            var dirTemp = Vector2.Normalize(goingTo);
+            var endPos = Position + (dirTemp * SpellOrigin.GetCurrentCastRange());
+
+            // usually doesn't happen
+            if (float.IsNaN(dirTemp.X) || float.IsNaN(dirTemp.Y))
+            {
+                if (float.IsNaN(CastInfo.Owner.Direction.X) || float.IsNaN(CastInfo.Owner.Direction.Y))
+                {
+                    dirTemp = new Vector2(1, 0);
+                }
+                else
+                {
+                    dirTemp = new Vector2(CastInfo.Owner.Direction.X, CastInfo.Owner.Direction.Z);
+                }
+
+                endPos = Position + (dirTemp * SpellOrigin.GetCurrentCastRange());
+                CastInfo.TargetPositionEnd = new Vector3(endPos.X, 0, endPos.Y);
+            }
+
+            // TODO: Verify if CastRangeDisplayOverride is the correct variable to use.
+            Destination = endPos;
+
+            ObjectsHit = new List<IGameObject>();
+        }
+
+        public override void Update(float diff)
+        {
+            if (!HasDestination() || _atDestination)
+            {
+                SetToRemove();
+                return;
+            }
+
+            if (!HasDestination())
```

Comment:
Useless `if`? It will enter the first if before.

Reply:
Leftover, removed.